os/drivers/compression/compress.c

```diff
@@ -0,0 +1,106 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <tinyara/fs/fs.h>
+#include <tinyara/fs/ioctl.h>
+#include <tinyara/compression.h>
+
+#include <errno.h>
+#include <debug.h>
+#include <assert.h>
+
+/****************************************************************************
+ * Function Prototypes
+ ****************************************************************************/
+
+static int comp_ioctl(FAR struct file *filep, int cmd, unsigned long arg);
+static ssize_t comp_read(FAR struct file *filep, FAR char *buffer, size_t len);
+static ssize_t comp_write(FAR struct file *filep, FAR const char *buffer, size_t len);
+
+/****************************************************************************
+ * Data
+ ****************************************************************************/
+static const struct file_operations compress_fops = {
+	0,                          /* open */
+	0,                          /* close */
+	comp_read,                  /* read */
+	comp_write,                 /* write */
+	0,                          /* seek */
+	comp_ioctl              /* ioctl */
+#ifndef CONFIG_DISABLE_POLL
+	, 0                         /* poll */
+#endif
+};
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+static ssize_t comp_read(FAR struct file *filep, FAR char *buffer, size_t len)
+{
+	return 0;
+}
+
+static ssize_t comp_write(FAR struct file *filep, FAR const char *buffer, size_t len)
+{
+	return 0;
+}
+
+
+/************************************************************************************
+ * Name: comp_ioctl
+ *
+ * Description: The ioctl method for compress.
+ *
+ ************************************************************************************/
+static int comp_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
+{
+	int ret = -ENOSYS;
+	struct compress_header *comp_info = (struct compress_header *)arg;
+
+	DEBUGASSERT(comp_info);
+	/* Handle built-in ioctl commands */
+	switch (cmd) {
+	case COMPIOC_COMPRESS:
+		ret = compress_block(comp_info->output_buffer, &comp_info->output_size, comp_info->input_buffer, comp_info->input_size);
+		break;
+	case COMPIOC_GET_COMP_TYPE:
+		/* CONFIG_COMPRESSION_TYPE 1 for LZMA and 2 for MINIZ */
+		comp_info->compress_type = CONFIG_COMPRESSION_TYPE;
```

Comment:
To get comp type (just 1 value), do we need to give entire `struct compres_header`?

Reply:
no , we can directly return the value . i will update the code